src/test/java/io/cdap/plugin/salesforce/plugin/source/batch/SalesforceBulkRecordReaderTest.java

```diff
@@ -313,6 +317,201 @@ private void assertRecordReaderOutputRecords(String[] csvStrings, Schema schema,
     }
   }
 
+  /**
+   * The FailsafeException is thrown with the expected error message.
+   * The retry policy is correctly configured with the specified configurations.
+   */
+  @Test(expected = FailsafeException.class)
```

Comment:
why `FailsafeException` here?

Javadoc comment suggests that `The FailsafeException is thrown with the expected error message`. Where are we comparing the expected error message?

Reply:
FailsafeException is expected as retry limit will get exceeded, changed the javadoc accordingly